### ISSUE TITLE
Publicize: LinkedIn: Support for Company Pages

### ISSUE THREAD

**Issue by kraftbj** [Automattic/jetpack#259]
Publicize currently only connects to your personal account; we should allow connections to Company Pages (http://marketing.linkedin.com/company-pages/) as well.

Ported from https://plugins.trac.wordpress.org/ticket/1727

**Comment by richardmtl** [comment]
Another request in 2056747-t

**Comment by kraftbj** [comment]
Requested in 2111488-t

**Comment by flamadiddle** [comment]
The landscape for LinkedIn's API has changed a lot in the past 2 years, but sharing to a Company page is possible (and appears it will remain so for the foreseeable future). Here are direct links to some relevant sections in the (REST) API docs:

 * [List Companies](https://developer.linkedin.com/docs/company-pages#list_companies)
 * [Company Share](https://developer.linkedin.com/docs/company-pages#company_share)

**Comment by oojacoboo** [comment]
@jeherve any update on this?

**Comment by jeherve** [comment]
@oojacoboo I don't have any news yet, but we'll post here as soon as we start making progress on this.

**Comment by kraftbj** [comment]
Requested in 2126804-t

**Comment by mikeyarce** [comment]
Requested in 2134159-t

**Comment by divyadeepsawhney** [comment]
Please add this feature asap. I am also a PHP developer but I never contributed to any project online. I am ready to help if anyone can guide me.

**Comment by flamadiddle** [comment]
@divyadeepsawhney, you could do a [pull request](https://help.github.com/articles/using-pull-requests/) on the [publicize module](https://github.com/Automattic/jetpack/tree/master/modules/publicize). Be sure to check out Jetpack's [project-specific instructions](http://jetpack.me/contribute/#patch) too. I'd do it myself (as I badly need this feature as well), but I simply have no time right now.

**Comment by jeherve** [comment]
Also requested here:
https://wordpress.org/support/topic/publicize-to-linkedin-company-page-3?replies=1&view=all

**Comment by jeherve** [comment]
Also suggested here:
https://wordpress.org/support/topic/how-to-connect-with-linkedin-company-page-instead-of-personal-one?replies=1&view=all

**Comment by jkudish** [comment]
Semi-related: #668

**Comment by jenhooks** [comment]
Requested here 2179590-t

**Comment by csonnek** [comment]
Requested in 2180984-t today.

**Comment by glenscott** [comment]
@aensley  It seems like a lot of the logic for sharing posts is within a web service call (https://public-api.wordpress.com/connect/?jetpack=publicize).  I don't think adding the LinkedIn Company page logic to https://github.com/Automattic/jetpack/tree/master/modules/publicize makes sense, unless I'm misunderstanding something?

**Comment by jkudish** [comment]
Correct, an addition like this would require an update on the WP.com side not only in Jetpack.

**Comment by mikeyarce** [comment]
Requested in 2203763-t

**Comment by mikeyarce** [comment]
Requested in 2206512-t

**Comment by csonnek** [comment]
Requested in 2213198-t

**Comment by sdavids13** [comment]
I would greatly appreciate this feature as well.

**Comment by JacobJWalker** [comment]
This feature would help our school tremendously!  I hope it is added as soon as possible.

**Comment by chaselivingston** [comment]
Also suggested in 2238951-t

**Comment by chaselivingston** [comment]
2239754-t

**Comment by kraftbj** [comment]
2245141-t

**Comment by annezazuu** [comment]
2251608-t

**Comment by csonnek** [comment]
2258798-t

**Comment by cromulus** [comment]
:+1:

**Comment by jeherve** [comment]
Also suggested here:
https://wordpress.org/support/topic/jetpack-share-with-linkedin-auto-chooses-my-personal-profile?replies=3&view=all

**Comment by csonnek** [comment]
2281239-t

**Comment by tareq1988** [comment]
:+1:

**Comment by annezazuu** [comment]
2302510-t

**Comment by divyadeepsawhney** [comment]
Hi. What's the development progress on this feature. I have some free time & I may contribute this feature. Kindly update me about the progress & instructions.

**Comment by jeherve** [comment]
@divyadeepsawhney We haven't made any progress on this yet. I'm afraid you won't be able to contribute to get this feature added to Jetpack, as most of the Publicize lives on WordPress.com servers, and not on your site or in this GitHub repo.

We'll post here as soon as we make progress on this.

**Comment by csonnek** [comment]
#2309461-t

**Comment by martinschenk** [comment]
would be great to get this feature.
Exists any other plugin which is able to do that?

**Comment by ntpixels** [comment]
#2362286-t

**Comment by csonnek** [comment]
Also in https://wordpress.org/support/topic/linkedin-url-not-correct?replies=1

**Comment by jeherve** [comment]
Also suggested in 2433382-t

**Comment by L34XXOR** [comment]
I also want this one :(

**Comment by mikakaltoft** [comment]
Ditto!

**Comment by lschuyler** [comment]
Also in 2460137-t

**Comment by ntpixels** [comment]
Also in 2462176-t

**Comment by iaikart** [comment]
I am also waiting for this..

**Comment by Korvacs** [comment]
This has been open for almost 2 years, has any consideration been given to it yet?

**Comment by jeherve** [comment]
@Korvacs We haven't made any progress on this yet. We'll update this issue as soon as we start working on it.

**Comment by annezazuu** [comment]
Requested in  2483159-t

**Comment by jpen365** [comment]
Would love to see this feature added

**Comment by mookid8000** [comment]
would be lovely! - :+1: +1

**Comment by lschuyler** [comment]
Requested in 2511680-t

**Comment by pabloleone** [comment]
same here, please!

**Comment by DennysDionigi** [comment]
same here too!
Thanks

**Comment by AlejandroGiardino** [comment]
same here too!
Thanks

**Comment by aensley** [comment]
I don't mean to seem impatient, but how many users have to request something before it becomes a priority? This has been open since [almost 3 years](https://plugins.trac.wordpress.org/ticket/1727) with nearly 50 different requests documented here and no progress. Were there really that many people asking for a Skype sharing button or yet another social icons widget?

The API calls are simple ( https://github.com/Automattic/jetpack/issues/259#issuecomment-76062603 ), and many have stepped up willing to do the work, except that it has to happen in a place we can't access.

Please make this a priority or tell us how we can help you do that. I'll start a social media campaign for it if you want. :-)

**Comment by lannison** [comment]
As there is still no progress, can anyone at least offer a workaround? Thanks

**Comment by jeherve** [comment]
@lannison Until this is implemented in Jetpack, you could try [this plugin](https://hu.wordpress.org/plugins/social-networks-auto-poster-facebook-twitter-g/). It seems to offer the option to post to LinkedIn company pages:

> LinkedIn - Autopost to your account. Ability to attach your blogpost to LinkedIn post. Autopost to LinkedIn Company pages and/or Groups (with third party API library)

**Comment by aensley** [comment]
That plugin requires a premium ["API" plugin](http://www.nextscripts.com/snap-api/) in order to enable that feature, which means we can't copy the code into our own solutions :-(. I may just have to write my own plugin for this one feature. If I do, I'll post a link here.

**Comment by aheckler** [comment]
2597682-t

**Comment by ntpixels** [comment]
2598964-t

**Comment by aheckler** [comment]
#2634407-t

**Comment by ntpixels** [comment]
#2653520-t

**Comment by Piyush5767** [comment]
Hi 
Sorry for late response but that is possible with linkedln api an i am currently using that feature in my web app through which i can post my content into linkedln business page.

I am also attach my script file and my script so that it would be easy for your to take a look and if you need live example then i will share you screen shots or my web app link where it is working. But before 
posting that link please check below files and script and if there is any issue then update me i will check that.

Where as is there any way to post content on google+ business page as same as linkedln web page?

/*------------------------------------ Script ------------------------------------*/
function linkedln_companypost(){
		$comment = $this->input->post('xxxxxxxx');;
		$title = $this->input->post('xxxxxxxxx');
		if($this->input->post('xxxxxxxxxxx') !='' && strlen($this->input->post('xxxxxxxxx'))>250){
			$description = substr($this->input->post('xxxxxxxxxxx'),0,250).'...';
		}else{
			$description = $this->input->post('xxxxxxxxxxx');
		}		
		$submitted_url = $this->input->post('xxxxxxxxx');
		$strShare = 
		'<?xml version="1.0" encoding="UTF-8"?> 
		<share>  
		  <comment>'.$comment.'</comment>  
		  <content> 
			 <title>'.$title.'</title>  
			<description>'.$description.'</description> 
			<submitted-url>'.$submitted_url.'</submitted-url> 
			<submitted-image-url>http://example.com</submitted-image-url> 
		  </content> 
		  <visibility> 
			<code>anyone</code> 
		  </visibility>  
		</share>';
		$access_token = 'your access token';
		$url = 'https://api.linkedin.com/v1/companies/7786995/shares?oauth2_access_token='.$access_token;
		// build your message
		$ch = curl_init($url);
		curl_setopt($ch, CURLOPT_POST, true);
		curl_setopt($ch, CURLOPT_POSTFIELDS,$strShare);
		curl_setopt($ch, CURLOPT_HEADER, 1);
		curl_setopt($ch, CURLOPT_POST, 1);	
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
		curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-type: application/xml','Connection: Keep-Alive'));
		$response = curl_exec($ch);
		$http_status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
		curl_close($ch);	
		if($http_status == 201){
			echo true;
		}else{
			echo false;
		}		
	}

**Comment by evelynriossf** [comment]
I would like to add my voice to a request for this much-needed feature. We need to be able to use the Jetpack Social Media Icons to link to Linkedin business accounts on business websites!

**Comment by Piyush5767** [comment]
Hi what about Google+ page ?

**Comment by nemchik** [comment]
Google+ pages already work with this plugin. If you're having a problem you should open a separate issue.

**Comment by aensley** [comment]
@evelynriossf This issue is about using the Publicize plugin to share links to LinkedIn Company pages. Your post - while helpful - is about the Social Media Icons Widget. That's a separate issue.

**Comment by evelynriossf** [comment]
My mistake. You are correct. It is a very similar issue for a separate Jetpack function.

**Comment by jeherve** [comment]
Also suggested here:
https://wordpress.org/support/topic/linkedin-support-1?replies=1&view=all

**Comment by annezazuu** [comment]
requested here 2686724-t

**Comment by ntpixels** [comment]
requested in 2700934-t

**Comment by walexus2000** [comment]
Please!!! Please!!! Make this a top priority!
Pleeease!!!!!!

**Comment by chaselivingston** [comment]
requested in 2704361-t

**Comment by samseen** [comment]
Common Automattic, I think it's high time you took this seriously.

**Comment by LordLiverpool** [comment]
+1 

[The Jetpack developers should watch this.](https://www.youtube.com/watch?v=NTSGp4UdEvQ)

**Comment by paramedicquebec** [comment]
I support this request!

**Comment by aensley** [comment]
To avoid repeating myself, I'll refer to my previous comment: https://github.com/Automattic/jetpack/issues/259#issuecomment-185819753

**Comment by lamdayap** [comment]
#2719453-t

**Comment by mstarnes** [comment]
I just read all of these +1 comments. There has to be more value in posting to company pages than personal pages. Could we at least get a reasonable status report on this much-requested feature?

**Comment by cena** [comment]
@2783446-t

**Comment by jeherve** [comment]
Also suggested here:
https://wordpress.org/support/topic/publish-automatically-on-my-company-page-linkedin?replies=1

**Comment by jrosseel** [comment]
+1

**Comment by jeherve** [comment]
Also suggested here:
https://wordpress.org/support/topic/cannot-connect-publicize-to-linkedin?replies=9#post-8786183

**Comment by naturalnetworks** [comment]
+1

**Comment by csonnek** [comment]
Also in 2819713-t

**Comment by vaishalijp** [comment]
I support this request!

**Comment by raphadko** [comment]
+1

**Comment by annezazuu** [comment]
2853698-t

**Comment by csonnek** [comment]
2860995-t

**Comment by sanand0** [comment]
+1

Actually, +10 -- if I could do that :-)

**Comment by benkalsky** [comment]
+1

**Comment by hisnameisjimmy** [comment]
+1

**Comment by laudesign** [comment]
+1

**Comment by albertonieto** [comment]
+1

**Comment by timmcdesign** [comment]
+1

**Comment by trsenna** [comment]
+1

**Comment by cedriccharles4** [comment]
Guys, what are you waiting for :p ???!!!
+10000

**Comment by newlyregistered26** [comment]
+1

**Comment by MitchTalmadge** [comment]
+1

**Comment by csonnek** [comment]
Requested in 2951321-t

**Comment by davidovich1** [comment]
Definitelly something I would really need. Thanks for any effort ...

**Comment by RCowles** [comment]
Requested in 2962930-t

**Comment by ChiefGyk** [comment]
Any news? My company could really use this, as we have one person dedicated to merely copying and pasting our articles

**Comment by wdac-davy** [comment]
One of our clients is pretty bummed out that this isn't a thing and posts don't auto-post to his company page. Any news on this?

**Comment by jeherve** [comment]
@ChiefGyk @wdac-davy We haven't made any progress on this yet. We'll update this issue as soon as we start working on it. Until this is implemented in Jetpack, you could try [this plugin](https://wordpress.org/plugins/social-networks-auto-poster-facebook-twitter-g/). It seems to offer the option to post to LinkedIn company pages:

> LinkedIn - Autopost to your account. Ability to attach your blogpost to LinkedIn post. Autopost to LinkedIn Company pages and/or Groups (with third party API library)

**Comment by davidcastrejon** [comment]
+1

**Comment by dac2020** [comment]
Thumbs DOWN on using NextScripts. We bought their solution and since we've started having problems posting to LinkedIn company pages, they won't respond to anyone. 

I (like many others I've seen posting) have both a ticket and a post in their support forum and all they have done in a week is to ask me to repeat myself. I've provided them with all the details including images in their support ticket system but they simply avoid me and others asking about this and similar problems.

Having this in wordpress, jetpack, what ever the politics are with Wordpress would be a plus after years of begging by countless people.

**Comment by LordLiverpool** [comment]
![jetpack](https://cloud.githubusercontent.com/assets/12895769/22399182/7b6ce458-e58f-11e6-9964-f656500edf65.jpg)

**Comment by FreeFXTradingStrategies** [comment]
So, just went through the process of setting up a LinkedIn company page. Took forever: email setup, images, etc. etc. Then went back to Publicize in Jetpack to link the new page and to my "chagrin" (who uses that word anymore), it would only link to my personal page. ARRRGGGGGHHHHHH!!!!

On the other hand, Facebook was simple. I have 6 company pages setup in Facebook and Publicize recognized them no problem and allowed me to connect just the one page that has to do with website at www.FreeFXTradingStrategies.com. AWESOME....

PLEASE make it where Linkedin company pages can be linked and not the personal profile, Jetpack. It would help greatly!!!

**Comment by dac2020** [comment]
Everyone is going to like hearing this. It isn't the solution but it's simple enough to do.
LinkedIn has now added an opengraph type option in their new company pages they are testing.
Just like on Facebook, you simply enter the url of the post and the image, title and text get picked up on the blog site.

While it isn't 100% automatic, for me at least, this solves the problem. 
I'm guessing this might also mean it will become a simpler solution for a plugin?

**Comment by lschuyler** [comment]
Also requested in 3028964-t

**Comment by marcuswallberg** [comment]
3 years later... Still not a priority?
+1

**Comment by lschuyler** [comment]
Requested in 3042716-t

**Comment by KraziyK** [comment]
I'd love to see this feature still.

**Comment by aensley** [comment]
![publicize_linkedin](https://cloud.githubusercontent.com/assets/95717/22832658/827804e6-ef74-11e6-91a4-c3ab539b7a6b.png)

You have **got** to be kidding me.

I give up. This will never happen, folks. Automattic simply doesn't care about this feature. Don't waste your time hoping like I have for the past **FOUR YEARS**.

**Comment by evelynriossf** [comment]
This is outrageous.

**Comment by schultzter** [comment]
Maybe we should start bugging LinkedIn about this. Perhaps they will have more luck convincing Automattic to add it to JetPack than we all do!!!

NOTE: I want it JetPack - I've already got enough trouble with all the plugins I have not working well together or even actively interfering with each other. I'm not installing another plugin just for this, my site's LinkedIn presence is at the mercy of Automattic.

**Comment by dac2020** [comment]
Since these guys don't seem to care for what the people want, I thought I should point out that LinkedIn *seems* to have made an update. I never noticed this before so am not sure if it's new or not but, they seem to have added opengraph import functions. 

When I went to my company page, I found that I could now simply enter the URL of the post I wanted to import and just like FB, the image, title and text excerpt got pulled in. That's my simple solution to this problem at this point and can even get rid of a plugin that's costing me fifty bucks a year.

Very tired of the subscription model, it *really* adds up to heavy costs.

**Comment by kraftbj** [comment]
Howdy y'all,

The Publicize Next milestone was one for an older initiative from another part of the company that did not include this enhancement. That project doesn't exist anymore and the removal of this from the milestone is meant to be reflective of reality (e.g. there is no "Publiclize Next" project).

We are very much aware of everyone here's desire for this feature and we are very mindful of it. We have been putting work into making this happen, unfortunately, there are a number of technical considerations that have made this much more complicated than it might seem at first glance. Thank you for your patience, and we look forward to sharing this feature with you as soon as possible.

**Comment by aensley** [comment]
@kraftbj If you're "putting work into making this happen," then logically, it should be assigned to a milestone. For the first time since this Issue was opened, it has none.

**Comment by kraftbj** [comment]
@aensley We're experimenting with moving away from using milestones for future versions and using it for project milestones instead, e.g. https://github.com/Automattic/jetpack/milestones . With our move to monthly releases, it isn't fair or accurate to assign PRs/issues to a monthly milestone unless it is ready to merge.

In other words, the lack of milestone is not indicative of potential for shipping or not with the system we're using now.

**Comment by georgestephanis** [comment]
Hi, I've been the developer working on adding LinkedIn Company Pages functionality.

As a partial technical explanation of what's been so tricky about building it out, a lot of it comes down to managing authentication.  The Publicize connection to LinkedIn was originally built out using a (then-current) OAuth1 implementation -- LinkedIn has since started offering a new OAuth2 implementation, which is what is required for using the endpoints and permissions to manage company pages.  The tricky bit is sorting out how to either manage two different authentication methods for the same service, or how to migrate the entire existing user base to the new authentication method without breaking their existing sites connections to LinkedIn.

There are some other technical considerations that aren't yet solved, but if LinkedIn allowed company pages to be managed through their legacy OAuth1 authentication implementation, we could have it shipped very quickly -- but they don't.

Long story short, the implementation is mostly completed, but we're trying to figure out how to roll it out without breaking everyone else's normal user LinkedIn connections, for the sake of a relative minority of business users.

**Comment by dac2020** [comment]
I suppose to real question is, why are there free and paid plugins that do this just fine? they don't seem to have problems and update as needed. the one we pay for is no longer working with linkedin company pages but I am sure they will update soon since it is a paid product. 

The point is, I would much rather consolidate our costs into one product that does most everything we need rather than having to dish out for dozens of costly plugins.

**Comment by georgestephanis** [comment]
@dac2020 As I just explained, a large part of it is legacy connections.  If we were just starting from scratch instead of having many users already connected via the OAuth1 method, it would have likely already been shipped.  We're working on solving this issue, but it's tricky -- it's not just Jetpack users, but every single WordPress.com blogger as well that has a LinkedIn connection.  If we suddenly broke all their connections, that would be far worse that holding off on shipping a new feature until we have a way to provide a good user experience for them as well!

**Comment by dac2020** [comment]
The usual way to solve that is to offer two options, one for those using an old method, one for those using a new method and explaining that they need to move to the new method and figuring out a good transition point so as not to need maintaining two options for good. 

It's how we do it at least.

**Comment by nemchik** [comment]
@georgestephanis Thank you for taking the time to explain the current situation. Obviously there is a decent number of people keeping track of this issue and hoping for an update. Just wanted to let you know that your comments here regarding the status of this issue and your involvement are very much appreciated and I wish you luck in finding a solution that meets the standards you wish to ship and I will cross my fingers for that to happen soon!

**Comment by aensley** [comment]
@georgestephanis Thank you! Thank you for explaining what's going on. More information about this task in the past 2 hours than in the past 2 years. Just a little more communication than "we're working on it" would have prevented a lot of frustration on our part.

As @dac2020 mentioned, the way our company would do this is in stages. Might I humbly suggest the following:

1. Deprecate the Oauth1 method.
2. Make all new connections (company or not) Oauth2, while retaining Oauth1 for existing connections.
3. Issue deprecation notice to Oauth1 users with removal date far down the road (6 months maybe). Urge them to re-authenticate their connection before that date.
4. Remove Oauth1 method on the specified date.

**Comment by georgestephanis** [comment]
Sure, that's a possibility, but rather awkward and kludgey.  Part of it is due to the large number of integrations with other parts of our codebase and external integrations, it's preferable to maintain the same identifier -- so running the two connections in parallel isn't exactly tenable, as then the new method would need to be named differently, and legacy systems wouldn't be able to use the new one.

Just please know that yes, we are working on it, but it is in no way the simple solution that some folks above have suggested.  If it were simple, it would be done.

**Comment by aensley** [comment]
@georgestephanis 

> Just please know that yes, we are working on it

We know that now, thanks to you. The previous years of correspondence haven't really communicated that to us.

* [10/24/14](https://github.com/Automattic/jetpack/issues/259#issuecomment-60438960) - I don't think LinkedIn has developed any API allowing automatic posting to company pages yet, but when they do we'll be sure to consider adding the option to Publicize.
* [3/7/15](https://github.com/Automattic/jetpack/issues/259#issuecomment-77727069) - I don't have any news yet, but we'll post here as soon as we start making progress on this.
* [9/16/15](https://github.com/Automattic/jetpack/issues/259#issuecomment-131517845) - We haven't made any progress on this yet.
* [1/8/16](https://github.com/Automattic/jetpack/issues/259#issuecomment-169971647) - We haven't made any progress on this yet. We'll update this issue as soon as we start working on it.
* [12/20/16](https://github.com/Automattic/jetpack/issues/259#issuecomment-268311911) - We haven't made any progress on this yet. We'll update this issue as soon as we start working on it.
* [2/10/17](https://github.com/Automattic/jetpack/issues/259#issuecomment-278989941) - We look forward to sharing this feature with you as soon as possible.

Hopefully, your team can see how that's frustrating your users. I'm not trying to be mean. I'm just hoping to show you that the stall tactics aren't improving user relations.

**Comment by maverick85** [comment]
LinkedIn has provided the means for this for over 3 years now. The lack of development/support for this most-needed feature is unbelievable.

If you're worried about breaking connections send out a warning notice 3 months earlier, 2 months earlier, 1 month earlier, 15 days earlier, 1 week earlier, 2 and 1 days before, then do it.

Cause that thing of holding back a much requested (and needed) feature because of possible broken integrations can only apply if it's not planned. Sometimes it's something that must happen.
Maybe it will break many customers who have NO MAINTENANCE... but is any website supposed to be without supervision or maintenance? Are those inactive customers a good reason to hold back this? C'mon...

**Comment by RafaelDeJongh** [comment]
Any updates on this, such a feature would be really welcomed as now I have to install and use a complete different plugin just to have this feature.

**Comment by cianciabella** [comment]
@RafaelDeJongh what plugin are you using in place of jetpack to accomplish this?

**Comment by RafaelDeJongh** [comment]
@cianciabella 

We started using hootsuite for that as NextScripts: Social Networks Auto-Poster didn't really work that well for us.

But perhaps it might work for you 😉

**Comment by dac2020** [comment]
NextScripts is an odd bird. When LinkedIn stopped working, myself and countless others contacted support, posted in the support forums and did all we could to get a simple acknowledgment but nothing came. Hmm, sounds a bit familiar :).

I have been wondering about this whole thread... are the slow responses from the devs because maybe they want to make this a premium addon?

**Comment by jeherve** [comment]
@dac2020 To clarify, whether the feature is paid or free it takes the same amount of time to build. :)

You can find out more about the challenges we face with this feature here:
https://github.com/Automattic/jetpack/issues/259#issuecomment-279004935

Once we make a decision and find a way to solve those challenges, we will let everyone know here!

**Comment by dac2020** [comment]
Thing is, it is not clear why this is such a big problem for you when others have had plugins out doing this for a very long time. I think this is probably the main mystery :).

Yes, LinkedIn changed and changes things but these plugins keep up so?

**Comment by jeherve** [comment]
@dac2020 George described some of our biggest challenges in the comment I highlighted above, and replied to your questions in the comments that followed.
Here are the follow-up comments:
- https://github.com/Automattic/jetpack/issues/259#issuecomment-279006817
- https://github.com/Automattic/jetpack/issues/259#issuecomment-279024987
I see you've also made a few suggestions and shared how you do it at your company. Thanks for sharing your ideas.

Now the ball is in our court: we'll need to make a decision, solve those challenges, find a solution that works for existing and new users alike, without breaking anyone's connection if possible, and we'll need to prioritize those changes alongside [the other things we also have to work on](https://github.com/Automattic/jetpack/issues?utf8=%E2%9C%93&q=is%3Aopen%20).

That will happen, but we don't have an ETA yet. When we do, we will let everyone know, we promise! Until then, there really isn't much you can do to help. Some folks have suggested alternatives above (thank you all for that), so I can only point you towards their recommendations if you need this feature today. If those solutions aren't what you're looking for, I'm afraid I can only recommend that you wait a little longer.

Thank you.

**Comment by cromulus** [comment]
I haven't been paying all that much attention, however, it seems like the
concerns around breaking pre-existing installation base could be
substantially alleviated by simply creating a second "type" of linkedin
connection, which is a business account.  Old, personal, linkedin
connections are unaffected.

**Comment by csonnek** [comment]
Requested in 3115362-t

**Comment by RafaelDeJongh** [comment]
@cromulus

That indeed might not be a bad idea actually, like for example if an existent connection is laid with Linkedin then keep using the old connection and opt a message saying they could update their connection to the new one. For new members the old connection would just not be displayed/or be able to be used.

Something like that might indeed work, but I do understand the challenges that this requires!

**Comment by tyxla** [comment]
Also requested in 3133799-t

**Comment by nshemesh** [comment]
+1

**Comment by aensley** [comment]
Priorities. Was the Internet Defense League widget really that important to users?

**Comment by jeherve** [comment]
@aensley Things haven't changed much since my last comment:
https://github.com/Automattic/jetpack/issues/259#issuecomment-285458253

I'll now be closing comments on this issue; I believe we’ve all already discussed about possible solutions and issues around this feature; newer comments won’t add much to the conversation at this point. We’ll update the issue as soon as we make progress on this.

Thank you.

**Comment by lamdayap** [comment]
requested here 649587-zen

**Comment by csonnek** [comment]
Requested in 670122-zen

**Comment by ashercantrell** [comment]
Requested in 626817-hc

**Comment by madhattermattic** [comment]
Requested in 704393-zen

**Comment by csonnek** [comment]
Also in 797528-zen

**Comment by SiobhyB** [comment]
Also requested in 1280963-hc

**Comment by eduardozulian** [comment]
Also in 881207-zen

**Comment by sarahgreenlee** [comment]
Also in 1434426-HC

**Comment by adelineyaw** [comment]
Another request via chat: 1882608-hc
Follow up ticket: 974227-zen

**Comment by grvrulz** [comment]
Also requested in 2423520-hc

**Comment by csonnek** [comment]
Requested in 1094834-zen

**Comment by scosgro** [comment]
Requested in 3573895-hc

**Comment by eduardozulian** [comment]
4661102-hc

**Comment by ntpixels** [comment]
1276749-zen

**Comment by brezocordero** [comment]
1279395-zen

**Comment by ntpixels** [comment]
1348046-zen

**Comment by kriskorn** [comment]
1396247-zen

**Comment by danjjohnson** [comment]
1514829-zen

**Comment by bikedorkjon** [comment]
1638587-zen

**Comment by coder-karen** [comment]
1662245-zen

**Comment by htdat** [comment]
1664696-zen

**Comment by SiobhyB** [comment]
1729662-zen

**Comment by formosattic** [comment]
9957029-hc

**Comment by RCowles** [comment]
1795507-zen

**Comment by KokkieH** [comment]
https://en.forums.wordpress.com/topic/publicize-my-content-to-my-buisness-page-on-linkedin-not-my-personal-page/

**Comment by darnelldibbles** [comment]
Requested here: https://twitter.com/jbmanos/status/1110651132802752512

**Comment by supernovia** [comment]
Requested here, too: https://twitter.com/ryonbrubaker/status/1131618706482782208

**Comment by ntpixels** [comment]
2117657-zen

**Comment by juliasydnor** [comment]
requested: 2185574-zen

**Comment by csonnek** [comment]
requested in 2201308-zen

**Comment by htdat** [comment]
2297085-zen

**Comment by csonnek** [comment]
Requested in 2358165-zen

**Comment by donpark** [comment]
It took me longer than expected (I keep underestimating how distracting GM can be) to dig into this issue but I have bad news: I don't think Publicize to LinkedIn Company Pages is going to work. Same for Publicize to LinkedIn Group Pages.

**LinkedIn Developer Program** is going through a lot of turmoil and creating confusion. It's docs were bad to begin with but is now in a shamble and convoluted.

**LinkedIn Groups API** is being deprecated (limited to approved partnership here means kept alive just to meet contract requirements) and related developer program is being shutdown after mixed result and rethinking after Microsoft purchase.

**Company API v1**, which used to work and was used in the 2017 GM Hack (D7277-code which no longer works), is deprecated.

**Company API v2** requires manual App Review that takes 20~30 days **per site**. Elaborate UI flow and stress on support goes well beyond that of simple OAuth2 flow which the Publicize feature is architected around. You can get a glimpse of what's needed in documentation for [WooCommerce Social Login plugin](https://docs.woocommerce.com/document/woocommerce-social-login-create-social-apps/#section-5).

**Conclusion**:

I don't think supporting LinkedIn Company Pages through Publicize feature makes sense. If support is a must then repackaging WooCommerce Social Login plugin as a bundled Jetpack module makes more sense.

cc @lancewillett for final say
cc @mdawaffe for his two cents since he rewrote much of Publicize support for LinkedIn v2 API.

**Comment by nickpagz** [comment]
Requested in 16512840-hc

**Comment by joweber123** [comment]
Requested in 16776448-hc

**Comment by jenhooks** [comment]
2566883-zen

**Comment by iamgabrielma** [comment]
2624507-zen

**Comment by csonnek** [comment]
2777900-zen

**Comment by GeoJunkie** [comment]
This came up today during a quick start session: 2813877-zen

**Comment by kosiew** [comment]
#19974774-hc

**Comment by htdat** [comment]
2849671-zen

**Comment by jeffgolenski** [comment]
2901200-zen

**Comment by lancewillett** [comment]
Worth revisiting this issue as it comes up fairly often in customer interactions.

CC @davemart-in for consideration on Manage team roadmap — maybe combined with any other LinkedIn API work that comes up for a sprint?

**Comment by davemart-in** [comment]
Thanks for the ping @lancewillett. I've added it to our list of project ideas.

**Comment by chaitanyamsv** [comment]
20858951-hc

**Comment by adelineyaw** [comment]
21341868-hc

**Comment by khristiansnyder** [comment]
This came up in chat today. 21479030-hc

Would be a great feature for our customers to have this set up.

**Comment by tmmbecker** [comment]
3106149-zen

**Comment by tmmbecker** [comment]
3110764-zen

**Comment by xpurichan** [comment]
Got a request for this in 22295103-hc.

**Comment by supernovia** [comment]
Another request here: https://wordpress.com/forums/topic/publisize-to-linkedin-company-page/

**Comment by jartes** [comment]
Another one here 24484517-hc

**Comment by snowmads** [comment]
Requested in 3398864-zen

**Comment by ktyfuller604** [comment]
Requested here - 23334137-hc

**Comment by rosepajaroja** [comment]
Requested here ZD-3526170

**Comment by Nic-Sevic** [comment]
another instance here 27863457-hc

**Comment by aamills** [comment]
another conversation about it 29312975-hc

**Comment by muffinpeace** [comment]
Requested on zen-4903092.

**Comment by formosattic** [comment]
+1 34927357-hc

**Comment by jromales** [comment]
Another instance in 35542722-hc

**Comment by ha-un** [comment]
Another request in 35283864-hc

**Comment by danmaze** [comment]
Another request in https://wordpress.org/support/topic/linkedin-page/

**Comment by bobmatyas** [comment]
Another request here https://wordpress.org/support/topic/new-version-of-facebook-page-cannot-detect/?view=all

**Comment by tellisbethel** [comment]
Another request here: 2704953-hc

**Comment by bobmatyas** [comment]
Requested here: https://wordpress.org/support/topic/linkedin-business-page/?view=all

**Comment by filipanoscampos** [comment]
Another such request: 5844394-zd-woothemes

**Comment by danjjohnson** [comment]
Another one: https://wordpress.org/support/topic/can-jetpack-publish-a-post-to-linkedin-group-or-company/

**Comment by bobmatyas** [comment]
Requested here: 5922507-zen

**Comment by jenhooks** [comment]
5947561-zen

**Comment by jartes** [comment]
41281511-hc

**Comment by bobmatyas** [comment]
Requested on 6252366-zen

**Comment by formosattic** [comment]
It was a deal breaker for this user 6270349-zd-woothemes.

**Comment by formosattic** [comment]
https://wordpress.org/support/topic/linkedin-pages/

**Comment by davemart-in** [comment]
Hi @pablinos, just wanted to check in and see if this has come across your team's radar yet. Thanks!

**Comment by pablinos** [comment]
It has, but we haven't prioritised it yet. The LinkedIn connection would probably need some additional refactoring to support this.

It would be nice to add this and Facebook Groups.

**Comment by formosattic** [comment]
+1 https://wordpress.org/support/topic/connecting-linkedin-business-page/

**Comment by deancroyal** [comment]
7142910-zd-a8c

**Comment by pehaa** [comment]
another request 7073999-zd-a8c

**Comment by bobmatyas** [comment]
Another request in the forums: https://wordpress.org/support/topic/social-share-to-business-page-vs-personal-page/

**Comment by bobmatyas** [comment]
Requested on 8023261-zen

**Comment by pehaa** [comment]
Another request - 8159460-zd-a8c

**Comment by bobmatyas** [comment]
Requested on 9019595-zen

**Comment by pablinos** [comment]
We now support this. There is a known issue with the caching of credentials for 5 minutes, if LinkedIn as previously been connected https://jetpack.com/support/jetpack-social/linkedin/#linkedin-company-pages but otherwise you can now connect and share to, company pages.
